src/deno_json/mod.rs

```diff
@@ -620,111 +611,13 @@ pub struct ConfigFileJson {
-/// ```
-fn decorate_tasks_json(
-  text: &str,
-  tokens: &[jsonc_parser::tokens::TokenAndRange<'_>],
-  comments: &HashMap<usize, std::rc::Rc<Vec<jsonc_parser::ast::Comment<'_>>>>,
-  tasks: &jsonc_parser::ast::Object,
-) -> Value {
-  let mut new_tasks = serde_json::Map::new();
-
-  // we want to exclude comments that aren't on their own line
-  // so the roundabout method here is to
-  // figure out whether there's a newline between the
-  // previous token and the comment in question
-
-  let task_tokens_start = tokens
-    .binary_search_by(|t| {
-      // we want the greatest token that is less than or equal to the start of the tasks
-      if t.range.end <= tasks.range.start {
-        std::cmp::Ordering::Less
-      } else {
-        std::cmp::Ordering::Greater
-      }
-    })
-    .unwrap_err();
-  let task_tokens_end = tokens[task_tokens_start..]
-    .iter()
-    .take_while(|t| t.range.end <= tasks.range.end)
-    .count()
-    + task_tokens_start;
-
-  let task_tokens = &tokens[task_tokens_start..task_tokens_end];
-
-  for prop in &tasks.properties {
-    let prop_comments =
-      comments.get(&prop.range.start).cloned().unwrap_or_default();
-
-    let mut comment_texts = Vec::with_capacity(prop_comments.len());
-
-    for comment in prop_comments.iter() {
-      let token_pos = task_tokens
-        .binary_search_by(|t| {
-          // we want to find the greatest token that is less than the start of comment
-          // (we can't search for token end == comment start because the
-          // preceding range might be a comment)
-          if t.range.end <= comment.range().start {
-            std::cmp::Ordering::Less
-          } else {
-            std::cmp::Ordering::Greater
-          }
-        })
-        .unwrap_err();
-      // the previous and next tokens
-      match (task_tokens.get(token_pos - 1), task_tokens.get(token_pos)) {
-        (Some(prev), Some(next)) => {
-          // on a different line than the previous and next tokens
-          if text[prev.range.end..comment.range().start].contains('\n')
-            && text[comment.range().end..next.range.start].contains('\n')
-          {
-            let comment_lines = comment.text().trim().split('\n').map(|s| {
-              s.trim().trim_start_matches('*').trim_start().to_string()
-            });
-            comment_texts.extend(comment_lines);
-          } else {
-            continue;
-          }
-        }
-        _ => continue,
-      };
-    }
-
-    new_tasks.insert(
-      prop.name.as_str().to_string(),
-      json!({
-        "comments": comment_texts,
-        "definition": Value::from(prop.value.clone()),
-      }),
-    );
-  }
-
-  Value::Object(new_tasks)
-}
-
 pub trait DenoJsonCache {
   fn get(&self, path: &Path) -> Option<ConfigFileRc>;
   fn set(&self, path: PathBuf, deno_json: ConfigFileRc);
 }
 
 #[derive(Clone, Debug, Default)]
-pub struct ConfigParseOptions {
-  pub include_task_comments: bool,
-}
+pub struct ConfigParseOptions {}
```

Comment:
I elected to keep this struct to not introduce a lot of changes to the signatures and in the future we might want to have other options.